Components/scripts/MapBase.gd

```diff
@@ -281,8 +284,58 @@ func wall(path):
 		changed_tiles[point + up] = true
 		changed_tiles[point + upup] = true
 		changed_tiles[point + downdown] = true
+	
+	return alreadyExistingWalls
+
+func possibleDoors(path):
+	var possibleDoorsInWalls = []
+	var possibleDoors = []
+	
+	if path.size() > 0:
+		var previousMovingHorizontal
+		
+		if (!is_corner_wall(path[0])):
+			possibleDoors.append(Vector2(path[0].x, path[0].y))
+		
+		if path.size() > 1:
+			for index in range(1, path.size()):
+				var point_a = path[index]
+				var point_b = path[index - 1]
+				var diff = point_b - point_a
+				var nextMoveHorizontal
+				
+				if diff[0] == 0 and diff[1] != 0:
+					nextMoveHorizontal = false
+				elif diff[0] != 0 and diff[1] == 0:
+					nextMoveHorizontal = true
+				else:
+					possibleDoorsInWalls.append(possibleDoors)
+					possibleDoors = []
+				
+				if previousMovingHorizontal == null && hasWallMoveChangedDirection(previousMovingHorizontal, nextMoveHorizontal) && possibleDoors.size() > 0:
+					possibleDoorsInWalls.append(possibleDoors)
+					possibleDoors = []
+				
+				previousMovingHorizontal = nextMoveHorizontal
+				
+				if (!is_corner_wall(point_a)):
+					possibleDoors.append(Vector2(point_a.x, point_a.y))
+	
+	possibleDoorsInWalls.append(possibleDoors)
+	
+	return possibleDoorsInWalls;
+
+func hasWallMoveChangedDirection(lastMoveHorizontal, nextMoveHorizontal):
+	return lastMoveHorizontal != nextMoveHorizontal
```

Comment:
Personally I think this function is more effort than it's worth, and think lastMove != nextMove is just as readable as the function name - just an opinion though :)

Reply:
Think i'll leave it, move can be quite ambiguous